packages/ui/v2/core/form/fields.tsx

```diff
@@ -246,7 +247,39 @@ export const PasswordField = forwardRef<HTMLInputElement, InputFieldProps>(funct
   props,
   ref
 ) {
-  return <InputField type="password" placeholder="•••••••••••••" ref={ref} {...props} />;
+  const { t } = useLocale();
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const toggleIsPasswordVisible = useCallback(
+    () => setIsPasswordVisible(!isPasswordVisible),
+    [isPasswordVisible, setIsPasswordVisible]
+  );
+  const textLabel = isPasswordVisible ? t("hide_password") : t("show_password");
+
+  return (
+    <div className="relative">
+      <InputField
+        type={isPasswordVisible ? "text" : "password"}
+        placeholder="•••••••••••••"
+        ref={ref}
+        {...props}
+        className={classNames("pr-10", props.className)}
+      />
+
+      <Tooltip content={textLabel}>
+        <button
+          className="absolute bottom-0 right-3 h-9 text-gray-900"
+          type="button"
+          onClick={() => toggleIsPasswordVisible()}>
+          {isPasswordVisible ? (
+            <EyeOff className="h-4 stroke-[2.5px]" />
+          ) : (
+            <Eye className="h-4 stroke-[2.5px]" />
+          )}
+          <span className="sr-only">{textLabel}</span>
+        </button>
+      </Tooltip>
```

Comment:
I believe we can just pass this into `addOnSuffix` prop that is attached to the input field to achieve the same functionality without needing "absolute" @JeroenReumkens

Reply:
Yeah I tried that. Only thing is the addons are squared off and have a divider line, so that would look quite different than the design. So not sure what @Jaibles thinks about that? 😄  See attached screenshot where the addon is used. Personally I like the design in Figma better.  

<img width="584" alt="Screenshot 2022-09-12 at 11 16 07" src="https://user-images.githubusercontent.com/2969573/189617409-e375e07a-2803-4233-9f9d-085825a50a83.png">
<img width="544" alt="Screenshot 2022-09-12 at 11 16 20" src="https://user-images.githubusercontent.com/2969573/189617464-cfdecdb2-c3f9-48df-9e91-2e065a15e01f.png">

Reply:
Definitely not ideal having the line for this.

Reply:
We could push a change to the
![CleanShot 2022-09-12 at 10 27 24](https://user-images.githubusercontent.com/55134778/189619860-50cce02b-5277-4443-bd1a-824b9a82e630.png)
the addonFilled option to set border-l-0 ? or create another prop I guess?

Reply:
Absolutely. I could definitely add another prop if that's the way to go. It does make the component a bit more complex by having all these different props 🤔  Personally I always try to stay away from that if possible. Composability is usually what I opt for rather that 'config' via props. That's why I was a bit hesitant to change this 😛  

At least by having the position absolute now we don't make the input component itself more complicate — so looking at it that way it might not be a very bad choice.

Reply:
After talking on slack - we've decided to stick with the current implementation due to the fact this design isn't used in the app that often. If this borderless suffix design keeps coming up we can think of implementing into this component or creating a wrapper like our `EmailInputs`

Awesome work!

Reply:
Agreed - not worth it in this instance